### PR TITLE
Keep original exceptions in limitreached

### PR DESCRIPTION
**Pull request by andreyrusanov**
@berislavlopac, @d0ugal this is a suggestion for #4.

If you are OK with idea I will add unit tests/update docs.

### CODE REVIEW

File: src/retrace_tests/test_basic.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def fn():
 
 
 def test_raises(keyboard_interrupt):
-    """Use the decorator and only retry on specific exceptions. The wrapped
+    """Use the decorator and only retry on specific exceptions_list. The wrapped
```
Comment (d0ugal) [inline src/retrace_tests/test_basic.py:29]: This looks like a bad search/replace.

File: src/retrace_tests/test_limits.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def fails_4_times_then_passes():
 
 
 def test_limit_passed_first_time(passes):
-    """Use the decorator and only retry on specific exceptions. The wrapped
+    """Use the decorator and only retry on specific exceptions_list. The wrapped
```
Comment (d0ugal) [inline src/retrace_tests/test_limits.py:28]: and again

File: src/retrace_tests/test_limits.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def fn():
 
 
 def test_limit_always_fails(fails):
-    """Use the decorator and only retry on specific exceptions. The wrapped
+    """Use the decorator and only retry on specific exceptions_list. The wrapped
```
Comment (d0ugal) [inline src/retrace_tests/test_limits.py:40]: here too

**Review by d0ugal — CHANGES_REQUESTED** [review]
Hey,

Sorry, I finally had a chance to test this. I do like the way it works, it feels pretty good. I'm not sure about the name `exceptions_list`. Would `caught_exceptions` explain it further?

This is how I tested it initially.

```
>>> import retrace
>>> 
>>> @retrace.retry()
... def fn():
...     raise Exception()
... 
>>> try:
...     fn()
... except retrace.LimitReached as e:
...     print e.exceptions_list
... 
[Exception(), Exception(), Exception(), Exception(), Exception(), Exception(), Exception(), Exception(), Exception(), Exception()]
```

However, using a custom function didn't work for me.

```
>>> import retrace
>>> 
>>> def limiter(attempt_number):
...     if attempt_number > 2:
...         raise retrace.LimitReached()
... 
>>> @retrace.retry(limit=limiter)
... def fn():
...     raise Exception()
... 
>>> try:
...     fn()
... except retrace.LimitReached as e:
...     print e.exceptions_list
... 
[]
```

From the issue...

> the only thing there that I missed is the case when somebody will try to add custom class as a limit an will not inherit it from ReasonsMixin. Some error should be raised, in this case, perhaps.

I think that is fine, if they don't use the mixin, they don't get the feature. We just need to document that. If we raise an error we would break anyone upgrading.

**Comment by andreyrusanov** [comment]
@d0ugal thanks for the review, I will work on it in a next few days.

**Pull request closed without merging**